### PR TITLE
Add property default show context value proeprty

### PR DESCRIPTION
**Pull request by Dzejkop**
This PR fixes (implements?) #97915

Add the `search.searchEditor.defaultShowContextValue` settings option to configure the default value for the number of context lines.

To test add
```
"search.searchEditor.defaultShowContextValue": 2
```

to settings, a new search editor should now show 2 lines of context.

`reusePriorSearchConfiguration` should take precedence though, so with settings like
```
"search.searchEditor.reusePriorSearchConfiguration": true,
"search.searchEditor.defaultShowContextValue": 2
```

new search editor windows should show the previously set number of context lines.

### CODE REVIEW

**Comment by JacksonKearl** [comment]
I wonder how this should interact with `reusePriorSearchConfiguration`. Currently if that is enabled this does nothing, I think this should possibly take precedence over the prior configuration.

**Comment by Dzejkop** [comment]
Makes sense to me.

If someone wants to have the old `reusePriorSearchConfiguration` they could just disable `defaultShowContextValue`. On the other hand with the current implementation if somebody wanted to reuse the prior config but override the default show context value - that wouldn't be possible.

File: src/vs/workbench/contrib/searchEditor/browser/searchEditorInput.ts [hunk 1]
```diff
@@ -304,11 +304,20 @@ export const getOrMakeSearchEditorInput = (
...
+
+	const reuseOldSettings = searchEditorSettings?.reusePriorSearchConfiguration;
+	const defaultShowContextValue = searchEditorSettings?.defaultShowContextValue;
+
 	const priorConfig: SearchConfiguration = reuseOldSettings ? new Memento(SearchEditorInput.ID, storageService).getMemento(StorageScope.WORKSPACE).searchConfig : {};
 	const defaultConfig = defaultSearchConfig();
+
 	let config = { ...defaultConfig, ...priorConfig, ...existingData.config };
 
+	if (defaultShowContextValue !== null && defaultShowContextValue !== undefined) {
```
Comment (JacksonKearl) [inline src/vs/workbench/contrib/searchEditor/browser/searchEditorInput.ts:317]: Changed this so that setting the property to 0 has an effect.

**Review by JacksonKearl — APPROVED** [review]
Looks good, I made one small change.

**Pull request merged**